src/monmove.cpp

```diff
@@ -1237,6 +1289,87 @@ bool monster::push_to( const tripoint &p, const int boost, const size_t depth )
     return true;
 }
 
+bool monster::take_item_at( const tripoint &p, std::function<std::list<item>::iterator(
+                                map_stack & )> selector )
+{
+    auto items = g->m.i_at( p );
+    auto to_pick_up = selector( items );
+
+    // TODO(sm): no volume/weight limits yet; AFAIK monsters/creatures have no notion of either
+    if( to_pick_up != items.end() ) {
```

Comment:
Would probably be easier to read if you just leave the function here (when `to_pick_up == items.end()`). Doing this avoids the indentation of the following code block.